src/cmds/tail.c

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file
+ * @brief tail-util
+ *
+ * @date 24.02.20
+ * @author Nishchay Agrawal
+ */
+
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <ctype.h>
+
+#define SIZE 200
+
+static void read_last_n_lines(FILE *fp, int n) {
+	int lines_checked = 0;
+
+	unsigned long long position_of_last_line_to_read;
+	char buf_str[2*SIZE];
```

Comment:
Just tried a big buffer size. 
In this, we could also do that we allocate the buffer just 1 byte larger than ftell(fp), but this might cause the array to ask for memory larger than the memory allocated to the function's stack if the file is very large.